app/objects/binary.py

```diff
@@ -80,8 +80,9 @@ def write_osu_string(self, string: str) -> BinaryWriter:
         """
         if string:
             self.buffer += b"\x0B"
-            self.write_uleb128(len(string))
-            self.write_raw(string.encode("utf-8"))
+            encoded_string = string.encode("utf-8")
+            self.write_uleb128(len(encoded_string))
+            self.write_raw(encoded_string)
```

Comment:
@realistikdash bug here - need to encode the string *before* fetching length, or non-ascii bytes will be counted incorrectly